### DISCUSSION TITLE
getStaticPaths, getStaticProps, next export, why does __NEXT_DATA__ need to be there?

### DISCUSSION THREAD

**Discussion by ironchef-bklyn** [vercel/next.js#24409] [Help] [answered]
We are using next.js to prerender all of our pages.  We use getStaticPaths to help us generate all of the routes/pages, we use getStaticProps to fetch all of the data that is needed to feed to React to create these pages.  We then use `next build` and `next export` to create the files.
We then copy these pages to an S3 bucket to be served up by our AWS services.

I've seen other posts asking about __NEXT_DATA__ being on the page.  And the answer was that it needs to be there for server side rendering.  We are not using any server side rendering...we literally are just serving up the generated pages.
Does __NEXT_DATA__ still have to be there for our scenario?  When I look a the generated page, all of the DOM elements are already there.

__NEXT_DATA__ seems to be just adding to the page size.

**Comment by devknoll** [comment, marked as answer]
Even when using `next export` to create your files, Next.js will still initialize and run React so that your components will run on the client to provide e.g. interactivity. To do this, React requires that you _hydrate_ the tree, by rendering the exact tree that the server rendered. In order to do that, Next.js needs the same input, which is passed in `__NEXT_DATA__`.


If you don't actually need any client side code, I think there's an experimental flag that you can turn on to strip that all out.

**Reply by tao** [reply]
Example of disabling client side code with the experimental config flag: https://github.com/vercel/next.js/discussions/22164#discussioncomment-367400
